src/renderer/system/queue.js

```diff
@@ -0,0 +1,18 @@
+import uuid from 'uuid';
+import { ipcRenderer as ipc } from 'electron';
+
+export function enqueue(channelName, fn, stack) {
+  const id = uuid.v4();
+
+  ipc.send('channel:enqueue', {
+    channelName,
+    stack,
+    id
+  });
+
+  ipc.once(`channel:execute:${id}`, () => {
```

Comment:
I know these are async, but I would feel much safer if this `once` listener was before the `send`.. 😅